scripts/cmake/vcpkg_find_acquire_program(NINJA).cmake

```
@@ -1,17 +1,29 @@
 set(program_name ninja)
-set(program_version 1.10.2)
+set(program_version 1.12.1)
 set(version_command --version)
 if(CMAKE_HOST_WIN32)
-    set(download_filename "ninja-win-${program_version}.zip")
-    set(tool_subdirectory "${program_version}-windows")
-    set(download_urls "https://github.com/ninja-build/ninja/releases/download/v${program_version}/ninja-win.zip")
-    set(download_sha512 6004140d92e86afbb17b49c49037ccd0786ce238f340f7d0e62b4b0c29ed0d6ad0bab11feda2094ae849c387d70d63504393714ed0a1f4d3a1f155af7a4f1ba3)
+    if(DEFINED ENV{PROCESSOR_ARCHITEW6432})
+        set(build_arch $ENV{PROCESSOR_ARCHITEW6432})
+    else()
+        set(build_arch $ENV{PROCESSOR_ARCHITECTURE})
+    endif()
+    if((build_arch MATCHES "^(ARM|arm)64$") OR (build_arch MATCHES "^(ARM|arm)$"))
```

```suggestion
    if(DEFINED ENV{PROCESSOR_ARCHITEW6432})
        set(HOST_ARCH $ENV{PROCESSOR_ARCHITEW6432})
    else()
        set(HOST_ARCH $ENV{PROCESSOR_ARCHITECTURE})
    endif()
    if(HOST_ARCH MATCHES "^(ARM|arm)64$")
```
 "host" is the term used in vcpkg. `HOST_ARCH` is the variable used for the other platforms here. Non-64 `arm` is not a relevant host arch for windows